Formula/awscli.rb

```diff
@@ -16,22 +27,117 @@ class Awscli < Formula
+  end
+
+  resource "prompt-toolkit" do
+    url "https://files.pythonhosted.org/packages/0c/37/7ad3bf3c6dbe96facf9927ddf066fdafa0f86766237cff32c3c7355d3b7c/prompt_toolkit-2.0.10.tar.gz"
+    sha256 "f15af68f66e664eaa559d4ac8a928111eebd5feda0c11738b5998045224829db"
+  end
+
+  resource "pycparser" do
+    url "https://files.pythonhosted.org/packages/0f/86/e19659527668d70be91d0369aeaa055b4eb396b0f387a4f92293a20035bd/pycparser-2.20.tar.gz"
+    sha256 "2d475327684562c3a96cc71adf7dc8c4f0565175cf86b6d7a404ff4c771f15f0"
+  end
+
+  resource "python-dateutil" do
+    url "https://files.pythonhosted.org/packages/4c/c4/13b4776ea2d76c115c1d1b84579f3764ee6d57204f6be27119f13a61d0a9/python-dateutil-2.8.2.tar.gz"
+    sha256 "0123cacc1627ae19ddf3c27a5de5bd67ee4586fbdd6440d9748f8abb483d3e86"
+  end
+
+  resource "ruamel-yaml" do
+    url "https://files.pythonhosted.org/packages/9a/ee/55cd64bbff971c181e2d9e1c13aba9a27fd4cd2bee545dbe90c44427c757/ruamel.yaml-0.15.100.tar.gz"
+    sha256 "8e42f3067a59e819935a2926e247170ed93c8f0b2ab64526f888e026854db2e4"
+  end
+
+  resource "s3transfer" do
+    url "https://files.pythonhosted.org/packages/27/90/f467e516a845cf378d85f0a51913c642e31e2570eb64b352c4dc4c6cbfc7/s3transfer-0.4.2.tar.gz"
+    sha256 "cb022f4b16551edebbb31a377d3f09600dbada7363d8c5db7976e7f47732e1b2"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/4f/5a/597ef5911cb8919efe4d86206aa8b2658616d676a7088f0825ca08bd7cb8/urllib3-1.26.6.tar.gz"
+    sha256 "f57b4c16c62fa2760b7e3d97c35b255512fb6b59a259730f36ba32ce9f8e342f"
+  end
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/25/9d/0acbed6e4a4be4fc99148f275488580968f44ddb5e69b8ceb53fc9df55a0/wcwidth-0.1.9.tar.gz"
+    sha256 "ee73862862a156bf77ff92b09034fc4825dd3af9cf81bc5b360668d425f3c5f1"
   end
 
   def install
     venv = virtualenv_create(libexec, "python3")
-    system libexec/"bin/pip", "install", "-v", "-r", "requirements.txt",
-                              "--ignore-installed", buildpath
-    system libexec/"bin/pip", "uninstall", "-y", "awscli"
+    remaining_resources = resources
+
+    # The `awscrt` package uses its own libcrypto.a on Linux. When building
+    # _awscrt.*.so, Homebrew's env seems to cause issues due to `openssl` flags.
+    # This causes the build to fail while running `scripts/gen-ac-index` with error:
+    # ImportError: _awscrt.cpython-39-x86_64-linux-gnu.so: undefined symbol: EVP_CIPHER_CTX_init
+    on_linux do
+      remaining_resources -= [resource("awscrt")]
+      resource("awscrt").stage do
+        with_env(CFLAGS: ENV.cflags, LDFLAGS: ENV.ldflags) do
+          ENV.prepend "CFLAGS", "-I#{Pathname.pwd}/build/deps/install/include"
+          ENV.prepend "LDFLAGS", "-L#{Pathname.pwd}/build/deps/install/lib"
+          system libexec/"bin/python3", *Language::Python.setup_install_args(libexec)
+        end
+      end
+    end
+
+    venv.pip_install remaining_resources
     venv.pip_install_and_link buildpath
```

Comment:
Another idea to consider. If we are okay with env modification applying to other pip instal; steps (should usually be a no-op unless directory exists), I found the following to be the simplest way to solve issue:
```rb
    on_linux do
      ENV.prepend "CFLAGS", "-I./build/deps/install/include"
      ENV.prepend "LDFLAGS", "-L./build/deps/install/lib"
    end
    virtualenv_install_with_resources
```

Reply:
I'm considering updating to this simpler installation after a day for deliberation. If there are any issues with some extra relative paths during compilation, let me know.

I'll add a revision bump at that time.